src/Dictionaries/HashedDictionary.cpp

```diff
@@ -10,7 +8,15 @@
 #include <Dictionaries//DictionarySource.h>
 #include <Dictionaries/DictionaryFactory.h>
 #include <Dictionaries/HierarchyDictionariesUtils.h>
+
+#include <Common/ArenaUtils.h>
```

Comment:
First we need to include boost, then Common, then Core, then Dictionaries.

Reply:
OK.

But just my two cents on this, **to me common headers should be included at the end**, since if you include it at the top, then you are explicitly include those headers to all the headers below (especially if those headers are never used anywhere else), and this will make header non self-compilable.